### PR TITLE
fixes #245

### LINKED ISSUE #245: remove unnecessary "use strict"
https://sideci.com has a list, otherwise all those that are used within a function that already has "use strict" - including private functions.

@Algareb or @Ardenian maybe?

### PR DESCRIPTION
**Pull request by Ardenian**
Seemed fine during testing, but requires re-formating, most likely.

### CODE REVIEW

File: src/mods/chat.js [hunk 1]
```diff
@@ -1,439 +1,440 @@
...
-                        tab.setAttribute ( 'class', tab.getAttribute ( 'class' ) + ' new-message' );
-                    } else if ( tabActive && chatActive && chat.lastChild.scrollTop < chat.lastChild.scrollHeight ) {
-                        try {
-                            chat.lastChild.scrollIntoView ( false );
-                        } catch ( e ) {
-                            idrinth.core.log ( e );
+                    for (var key in messages) {
+                        if (
+                                !isNaN ( parseInt ( key, 10 ) )
+                                && document.getElementById ( 'idrinth-chat-tab-' + key )
```
Comment (IdrinthGitbot) [inline src/mods/chat.js]: __[jshint]__
* Bad line breaking before '&&'.

File: src/mods/chat.js [hunk 2]
```diff
@@ -1,439 +1,440 @@
...
+                                && document.getElementById ( 'idrinth-chat-tab-' + key ).getElementsByTagName ( 'ul' )[1] ) {
```
Comment (IdrinthGitbot) [inline src/mods/chat.js]: __[jshint]__
* Bad line breaking before '&&'.

File: src/mods/chat.js [hunk 3]
```diff
@@ -694,270 +695,271 @@ idrinth.chat = {
...
+                headers.pass = chatLogin[1].value;
+                success = idrinth.chat.loginCallback;
+            }
+            idrinth.core.ajax.runHome ( urls[key], success, fail, timeout, JSON.stringify ( headers ) );
         }
-        idrinth.core.ajax.runHome ( urls[key], success, fail, timeout, JSON.stringify ( headers ) );
-    }
-};
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/chat.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/core.js [hunk 4]
```diff
@@ -1,360 +1,358 @@
...
+                            handleElement ( idrinth.ui.matchesCss ( event.target, selector ), event.type, selector );
+                        }
                     }
                 }
             }
         }
-    }
-};
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/core.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/facebook.js [hunk 5]
```diff
@@ -1,34 +1,37 @@
...
-        idrinth.facebook.popup = window.open ( "https://apps.facebook.com/dawnofthedragons/" );
-        idrinth.facebook.popup.onload = function () {
-            idrinth.core.timeouts.add ( 'facebook', idrinth.facebook.restart, 3333 );
-        };
-        // in case onload fails
-        idrinth.core.timeouts.add ( 'facebook', idrinth.facebook.restart, 11111 );
-    }
-};
\ No newline at end of file
+    };
```
Comment (IdrinthGitbot) [inline src/mods/facebook.js:36]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/land.js [hunk 6]
```diff
@@ -1,263 +1,266 @@
...
+                /**
+                 * @type Number
+                 */
+                base: 1200000
+            }
         }
-    }
-};
\ No newline at end of file
+    };
```
Comment (IdrinthGitbot) [inline src/mods/land.js:265]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/names.js [hunk 7]
```diff
@@ -1,269 +1,269 @@
...
-        };
-        if ( idrinth.platform === 'kongregate' ) {
-            idrinth.core.multibind.add ( 'mouseover', '.chat_message_window .username', showTooltip );
-            idrinth.core.timeouts.add ( 'names', idrinth.names.run, 10000 );
-            build ();
         }
-    }
-};
\ No newline at end of file
+    };
```
Comment (IdrinthGitbot) [inline src/mods/names.js:268]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/newgrounds.js [hunk 8]
```diff
@@ -1,62 +1,65 @@
...
+                        } catch ( e3 ) {
+                            //already gone, nothing to do
+                        }
+                        idrinth.raids.join.messages.trying ( key );
+                        idrinth.newgrounds.join ();
+                    },
+                    idrinth.settings.get ( "newgroundLoad" ) * 1000
+                    );
+        }
+    };
```
Comment (IdrinthGitbot) [inline src/mods/newgrounds.js:89]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/raids.js [hunk 9]
```diff
@@ -1,311 +1,303 @@
...
         }
-    },
-    start: function ( ) {
-        'use strict';
-        idrinth.core.timeouts.remove ( 'raids' );
-        idrinth.core.timeouts.add ( 'raids', idrinth.raids.join.process, 1500, -1 );
-    }
-};
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/raids.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/settings.js [hunk 10]
```diff
@@ -1,425 +1,425 @@
...
+                    getCurrent ();
+                } else {
+                    getOld ();
+                }
             }
         }
-    }
-};
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/settings.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/stats.js [hunk 11]
```diff
@@ -1,111 +1,114 @@
...
+                if ( result.hasOwnProperty ( key ) ) {
+                    document.getElementById ( 'idrinth-stats-' + key ).parentNode.parentNode.childNodes[2].innerHTML = result[key] < 0 ? result[key] : '+' + result[key];
+                    document.getElementById ( 'idrinth-stats-' + key ).value = Number.parseInt ( document.getElementById ( 'idrinth-stats-' + key ).value, 10 ) + result[key];
+                }
             }
         }
-    }
-};
\ No newline at end of file
+    };
```
Comment (IdrinthGitbot) [inline src/mods/stats.js:113]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/text.js [hunk 12]
```diff
@@ -1,73 +1,76 @@
...
-            }
-            return idrinth.text.data.default;
-        };
-        return getSub ( idrinth.text.data, key.split ( '.' ), getSub );
-    }
-};
+            return getSub ( idrinth.text.data, key.split ( '.' ), getSub );
+        }
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/text.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/tier.js [hunk 13]
```diff
@@ -1,380 +1,382 @@
...
+                    result.push ( key );
+                }
             }
+            makeList ( result );
         }
-        makeList ( result );
-    }
-};
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/tier.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/ui.js [hunk 14]
```diff
@@ -1117,26 +1111,27 @@ idrinth.ui = {
...
                 } ]
-        } ) );
-        build ( );
-    }
-};
+            } ) );
+            build ( );
+        }
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/ui.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/user.js [hunk 15]
```diff
@@ -1,98 +1,100 @@
...
-            }
-            idrinth.core.ajax.runHome ( 'i-am-alive/' + getIdentifier () + '/' );
-        };
-        idrinth.core.timeouts.add ( 'user', sendAlive, 20000 );
-    }
-};
+            idrinth.core.timeouts.add ( 'user', sendAlive, 20000 );
+        }
+    };
+})
```
Comment (IdrinthGitbot) [inline src/mods/user.js]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/war.js [hunk 16]
```diff
@@ -1,318 +1,321 @@
...
-        build ();
-    }
-};
\ No newline at end of file
+                idrinth.ui.base.appendChild ( idrinth.war.element );
+            };
+            idrinth.core.timeouts.add ( 'war', idrinth.war.getData, 5000 );
+            build ();
+        }
+    };
```
Comment (IdrinthGitbot) [inline src/mods/war.js:337]: __[jshint]__
* Expected an assignment or function call and instead saw an expression.

File: src/mods/chat.js [hunk 17]
```diff
@@ -1,440 +1,441 @@
...
-                    } else if ( tabActive && chatActive && chat.lastChild.scrollTop < chat.lastChild.scrollHeight ) {
-                        try {
-                            chat.lastChild.scrollIntoView ( false );
-                        } catch ( e ) {
-                            idrinth.core.log ( e );
+                    for ( var key in messages ) {
+                        if (
+                                !isNaN( parseInt( key, 10 ) )
+                                && document.getElementById( 'idrinth-chat-tab-' + key )
+                                && document.getElementById( 'idrinth-chat-tab-' + key ).getElementsByTagName( 'ul' )[1] ) {
```
Comment (IdrinthGitbot) [inline src/mods/chat.js]: __[jshint]__
* Bad line breaking before '&&'.

File: src/mods/chat.js [hunk 18]
```diff
@@ -1,963 +1,964 @@
...
+                        return;
+                    }
+                    var translation = idrinth.text.get ( node.label );
+                    return {
+                        content: translation === idrinth.text.data.default ? node.label : translation,
+                        type: 'li',
+                        attributes: [ {
+                                name: 'onclick',
+                                value: 'idrinth.chat.useroptions(' + node.chat + ',' + user + ',\'' + node.rank + '\');'
+                                        + 'this.parentNode.parentNode.removeChild(this.parentNode);'
```
Comment (IdrinthGitbot) [inline src/mods/chat.js]: __[jshint]__
* Bad line breaking before '+'.

**Comment by w20k** [comment]
@Ardenian  could you try removing this issue - "Rule 'no-empty-label' was removed and replaced by: no-labels (no-empty-label)", in codacy.

**Comment by Idrinth** [comment]
it's just an issue in that copied version, @w20k , not in the official version.

**Comment by Idrinth** [comment]
more importantly, @Ardenian , please merge in the master and fix the conflicts

File: src/mods/core.js [hunk 19]
```diff
@@ -1,382 +1,385 @@
...
+             * @param {string} url
+             * @param {function} success
+             * @param {function} failure
+             * @param {function} timeout
+             * @param {string} additionalHeader
+             * @param {Boolean} [false] isStatic
+             * @returns {undefined}
+             */
+            run: function ( url, success, failure, timeout, additionalHeader ) {
+                'use strict';
```
Comment (IdrinthGitbot) [inline src/mods/core.js]: __[jshint]__
* Unnecessary directive "use strict".

File: src/mods/core.js [hunk 20]
```diff
@@ -1,382 +1,385 @@
...
+             * @type {object}
+             */
+            list: { },
+            /**
+             * makes sure the timeout is removed when called next
+             * @param {string} identifier
+             * @returns {undefined}
+             */
+            remove: function ( identifier ) {
+                'use strict';
```
Comment (IdrinthGitbot) [inline src/mods/core.js]: __[jshint]__
* Unnecessary directive "use strict".

File: src/mods/core.js [hunk 21]
```diff
@@ -1,382 +1,385 @@
...
+            /**
+             *
+             * @param {string} identifier
+             * @param {function} func
+             * @param {int} time in milliseconds
+             * @param {Number} [1] maxRepeats -1 will repeat until the end of time
+             * @returns {undefined}
+             */
+            add: function ( identifier, func, time, maxRepeats ) {
+                'use strict';
```
Comment (IdrinthGitbot) [inline src/mods/core.js]: __[jshint]__
* Unnecessary directive "use strict".

File: src/mods/core.js [hunk 22]
```diff
@@ -1,382 +1,385 @@
...
+                if ( !idrinth.core.timeouts.next ) {
+                    idrinth.core.timeouts.next = window.setTimeout ( idrinth.core.timeouts.process, 1000 );
+                }
+            },
+            /**
+             * activates all relevant timeouts and intervals
+             * @returns {undefined}
+             */
+            process: function () {
+                'use strict';
```
Comment (IdrinthGitbot) [inline src/mods/core.js]: __[jshint]__
* Unnecessary directive "use strict".

File: src/mods/core.js [hunk 23]
```diff
@@ -1,382 +1,385 @@
...
+                }
+            }
+        },
+        /**
+         *
+         * @param {string} string
+         * @returns {undefined}
+         */
+        log: function ( string ) {
+            'use strict';
```
Comment (IdrinthGitbot) [inline src/mods/core.js]: __[jshint]__
* Unnecessary directive "use strict".

File: src/mods/tier.js [hunk 24]
```diff
@@ -1,387 +1,418 @@
...
+                }
+            }
+            idrinth.core.alert ( idrinth.text.get ( "tier.maxBoxes" ) );
+        },
+        /**
+         * initializes this module
+         * @returns {undefined}
+         */
+        start: function ( ) {
+            'use strict';
```
Comment (IdrinthGitbot) [inline src/mods/tier.js]: __[jshint]__
* Unnecessary directive "use strict".

File: src/stable.js [hunk 25]
```diff
@@ -1,101 +1,103 @@
...
+                return idrinth.core.timeouts.add ( 'start', startInternal, 2222 );
+            }
+            startInternal ();
         }
-        startInternal ();
-    }
-};
-window.setTimeout ( idrinth.start, 6666 );
+    };
+    window.setTimeout ( idrinth.start, 6666 );
```
Comment (IdrinthGitbot) [inline src/stable.js:97]: __[jshint]__
* Unreachable 'window' after 'return'.

**Comment by Idrinth** [comment]
I think I fixed the stuff now

**Comment by Idrinth** [comment]
LGTM ,testing worked out, @w20k
